tests/test_requirements.py

```diff
@@ -309,6 +309,11 @@ def test_parse_requirements_stress(chroot):
             ),
         )
     )
+
+    # Ensure local non-distribution files matching distribution names are not erroneously probed
+    # as distributions to find name and version metadata.
+    touch("nose")
```

Comment:
One thing I appreciate about your PEX tests is explicitly mentioning the motivating issue in things like test names. Consider adding a link to the issue in this comment.

(Not a huge deal - Git history will trace it back.)

Reply:
Yeah - its a cheap / sneaky / convenient ' lazy test. Agreed about git history though. Now that the black reformat is past, git blame is pretty useful for exactly this.